cni/pkg/nodeagent/server.go

```diff
@@ -111,10 +157,14 @@ func NewServer(ctx context.Context, ready *atomic.Value, pluginSocket string, ar
 	}
 
 	// Create hostprobe rules now, in the host netns
-	hostIptables.DeleteHostRules()
-
-	if err := hostIptables.CreateHostRulesForHealthChecks(); err != nil {
-		return nil, fmt.Errorf("error initializing the host rules for health checks: %w", err)
+	if err := runAsNs(hostNetNS, func() error {
+		hostIptables.DeleteHostRules()
+		if err := hostIptables.CreateHostRulesForHealthChecks(); err != nil {
```

Comment:
Ditto - it's called `hostside`, let's take care of the host stuff inside

(also we should hold the host netns FD as briefly as possible, like for closure-duration - not as top-level state)